_posts/2023-06-05-timelines-save-time-and-money.md

```diff
@@ -0,0 +1,294 @@
+Conceptually, that is enough information to do the calculations we want. Kaskada
+knows how to get it done without us having to specify (yet again) what a day is,
+how sliding windows work, or how to manipulate the rows of event data to make it
+happen. With less code and less logic to worry about, there is much less surface
+area to introduce bugs or inefficiencies.
+
+
+## We shouldn’t have to write query logic about how time works
+
+To put it half-jokingly, [Kaskada understands
+time](https://kaskada.io/2022/12/15/the-kaskada-feature-engine-understands-time.html),
+while with SQL you have to say repeatedly that the past comes before the future
+and that a calendar day still exists even if you don’t have a row for it. How
+many times in SQL do I have to write the same `JOIN` logic stating that the two
+entity IDs have to be the same and that the event timestamps need to be less
+than the time bucket/window timestamps? In Kaskada, you don’t have to write it
+at all, and there are many other bugs and inefficiencies that can be avoided by
+letting Kaskada handle time logic in the intuitive way (unless you want to use
+different logic).
+
+
+## Kaskada’s native timeline abstraction was built for this
+
+[Kaskada's concept of
+timelines](https://kaskada.io/2023/05/09/introducing-timelines-part-1.html) and
+its overall time-centric design means that natural logic around time
+calculations happens natively. Kaskada can do a 30-day rolling average with a
+single line of code, more or less, and you don't have to think about the rows
+filled with zeros or tables blowing up in size because the computational model
+is focused on progressing through time, and not on producing, manipulating, and
+storing rows.
+
+In contrast, as illustrated earlier in this article, SQL has at least two
+distinct ways to deal with time:
+
+* __window functions__: build one row per day so that row calculations become day calculations
+* __time JOINs__: event data JOINed into time buckets and aggregated
+
+The first is conceptually easy to understand and potentially easy to write in
+SQL if you are an expert. The second is not quite as easy to understand, and
+requires writing logic in SQL about what it means for an event to belong to a
+day or date. Both types of implementations leave the coder to choose how time
+should work on each and every query that needs it—a recipe that invites trouble.
+
+
+## Try it out and let us know what you think!
+
+Kaskada is open-source and easy to set up on your local machine in just a few
+minutes. For more information about how Kaskada and native timelines can
+dramatically simplify coding logic and computational efficiency for time-centric
+feature engineering and analytics on event-based data, see [our
+documentation](https://kaskada.io/docs-site/kaskada/main/overview/what-is-kaskada.html),
+in particular the [Getting Started
+page](https://kaskada.io/docs-site/kaskada/main/getting-started/index.html). 
+
+You can also [join our Slack
```

Comment:
Created a function for this since the link will expire periodically and we'll need to regenerate everything.

`[join our Slack channel]({{site.join_slack}})`